src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -835,6 +835,8 @@ private SecurityStatusPal GenerateToken(ReadOnlySpan<byte> inputBuffer, ref byte
 
         internal SecurityStatusPal Renegotiate(out byte[]? output)
         {
+            ArgumentNullException.ThrowIfNull(_securityContext);
```

Comment:
_securityContext isn't an argument.  In what situations will we get here and find _securityContext is null?

Reply:
It would happen if this is called before `AuthenticateAs*`. However, `NegotiateClientCertificateAsync` has guard `ThrowIfExceptionalOrNotAuthenticated`.  I can remove it or turn it into `Debug.Assert`

Reply:
updated to Debug.Assert